From a4e24f29df7daf277f3f81dde72ca89efb7f5ade Mon Sep 17 00:00:00 2001
From: Hans Rosenfeld <hans.rosenfeld@joyent.com>
Date: Wed, 25 Apr 2018 18:04:15 +0000
Subject: [PATCH] OS-6894 set passthru flag when ppt devices are assigned to a
 VM Reviewed by: Robert Mustacchi <rm@joyent.com> Reviewed by: Patrick Mooney
 <patrick.mooney@joyent.com> Reviewed by: Jerry Jelinek
 <jerry.jelinek@joyent.com> Approved by: Robert Mustacchi <rm@joyent.com>

---
 usr/src/uts/i86pc/io/vmm/io/ppt.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/usr/src/uts/i86pc/io/vmm/io/ppt.c b/usr/src/uts/i86pc/io/vmm/io/ppt.c
index d36c360e03..dabbe584fd 100644
--- a/usr/src/uts/i86pc/io/vmm/io/ppt.c
+++ b/usr/src/uts/i86pc/io/vmm/io/ppt.c
@@ -57,6 +57,7 @@ __FBSDID("$FreeBSD$");
 #include <sys/sunddi.h>
 #include <sys/pci.h>
 #include <sys/pci_cap.h>
+#include <sys/pcie_impl.h>
 #include <sys/ppt_dev.h>
 #include <sys/mkdev.h>
 #include <sys/sysmacros.h>
@@ -1019,6 +1020,7 @@ ppt_assign_device(struct vm *vm, int pptfd)
 	ppt->vm = vm;
 	iommu_remove_device(iommu_host_domain(), pci_get_bdf(ppt->pptd_dip));
 	iommu_add_device(vm_iommu_domain(vm), pci_get_bdf(ppt->pptd_dip));
+	pf_set_passthru(ppt->pptd_dip, B_TRUE);
 
 done:
 	releasef(pptfd);
@@ -1068,6 +1070,8 @@ ppt_do_unassign(struct pptdev *ppt)
 	ppt_reset_pci_power_state(ppt->pptd_dip);
 	(void) pci_restore_config_regs(ppt->pptd_dip);
 
+	pf_set_passthru(ppt->pptd_dip, B_FALSE);
+
 	ppt_unmap_mmio(vm, ppt);
 	ppt_teardown_msi(ppt);
 	ppt_teardown_msix(ppt);
-- 
2.21.0

